content/articles/2017-12-workflow-vs-script.md

```diff
@@ -0,0 +1,126 @@
+---
+title: workflow
+author: Jenny Bryan
+date: '2017-12-12'
+slug: workflow-vs-script
+description: >
+  Advice on workflows for developing R scripts. How to decide if an action belongs in the script or not.
+categories:
+  - programming
+photo:
+  url: https://commons.wikimedia.org/wiki/File:Burned_laptop_secumem_11.jpg
+  author: secumem
+tags: []
+---
+
+I gave a talk today at the [IASC-ARS/NZSA Conference](http://www.nzsa2017.com), hosted by the Stat Department at The University of Auckland. One of the conference themes is to celebrate the accomplishments of Ross Ihaka, who got R started back in 1992, along with Robert Gentleman. My talk included lots of advice on ways to set up your R life to maximize effectiveness and reduce frustration.
+
+Two specific slides have generated much discussion and consternation in #rstats Twitter and I think more explanation is in order:
+
+*insert suitably small images here or use quotation*
+
+> If the first line of your R script is
+> 
+> `rm(list = ls())`
+> 
+> I will come into your office and
+>      SET YOUR COMPUTER ON FIRE 🔥.
+
+> If the first line of your R script is
+> 
+> `setwd("C:\Users\jenny\path\that\only\I\have")`
+> 
+> I will come into your office and
+>      SET YOUR COMPUTER ON FIRE 🔥.
+
+
+Here's my attempt to explain *why* I have such strong opinions.
+
+## Workflow versus Product
+
+It's important to recognize the difference between things you do because of personal taste and habits ("workflow") versus the logic and output that is the essence of your project ("product"). Here are examples of what I consider to be workflow:
+
+  * The IDE you use, e.g., RStudio, Emacs with ESS, or nothing at all.
+  * The name of your home directory.
+  * The R code you ran before lunch.
+  
+Here are examples of what I consider to be product:
+
+  * The raw data.
+  * The R code I would need to run on your raw data to get your results, including the explicit `library()` calls to load necessary packages.
+  
+Ideally, no one can detect anything about your workflow from your product. Or at least, it should be as easy as possible for people with a different workflow to reproduce your product if they have your input and code. 
+
+## Self-sufficiency
+
+I suggest organizing data analyses into *projects*: a folder on your computer that holds all the files relevant to a particular piece of work. Any resident R script should assume that it will be run from a fresh R process with working directory set to the directory where this project lives. This convention guarantees that the project can be moved around on your computer or onto other computers and will still "just work". I argue that this is the only practical convention that creates reliable behavior across different computers or users and over time.
```

Comment:
I think this needs a parenthetical about "RStudio projects" just to emphasise that you're not implying RStudio usage here (I see that you come back to that later, but I think it needs at least a nod here)